server/src/inmantals/server.py

```diff
@@ -77,9 +103,10 @@ def __init__(self, instream: BaseIOStream, outstream: BaseIOStream, address):
         # compiler_venv_path is only relevant for versions of core that require a compiler venv. It is ignored otherwise.
         self.compiler_venv_path: Optional[str] = None
 
-    async def initialize(self, rootPath, rootUri, capabilities: Dict[str, object], **kwargs):  # noqa: N803
+    async def initialize(self, rootPath, rootUri, capabilities: Dict[str, object], useCache=True, **kwargs):  # noqa: N803
         logger.debug("Init: " + json.dumps(kwargs))
-
+        if not useCache:
+            Config.set("compiler", "cache", "False")
```

Comment:
It may be simpler to just pass `Project(attach_cf_cache=useCache)`. You'd also need to check when the cache was introduced and whether we need to do some backward compatibility tricks.

Side note: we use camelCase for these options here because that's how they're defined in the spec. For our own options I'd go for snake_case.